### PR TITLE
Enable Extension Support

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142028

cc @malfet @zou3519 @xmfan @ptrblck @msaroufim @eqy

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
lul

**Review by eqy — APPROVED** [review]

File: torch/utils/cpp_extension.py [hunk 1]
```diff
@@ -1991,10 +1991,12 @@ def _get_cuda_arch_flags(cflags: Optional[List[str]] = None) -> List[str]:
         ('Ampere', '8.0;8.6+PTX'),
         ('Ada', '8.9+PTX'),
         ('Hopper', '9.0+PTX'),
+        ('Blackwell', '10.0+PTX'),
     ])
 
     supported_arches = ['3.5', '3.7', '5.0', '5.2', '5.3', '6.0', '6.1', '6.2',
-                        '7.0', '7.2', '7.5', '8.0', '8.6', '8.7', '8.9', '9.0', '9.0a']
+                        '7.0', '7.2', '7.5', '8.0', '8.6', '8.7', '8.9', '9.0', '9.0a'
```
Comment (xwang233) [inline torch/utils/cpp_extension.py:1998]: FYI, missing a comma here, so the two strings are joined together.

Comment (drisspg) [reply torch/utils/cpp_extension.py:1998]: ughh, thanks for pointing this out was flying a little blind here

**Comment by Turakar** [comment]
This PR did not address the issue with 9.0a for cases where TORCH_CUDA_ARCH_LIST is not set. In this case, the code tries to parse torch.cuda.get_arch_list(), and fails:

```
  File ".venv/lib/python3.12/site-packages/torch/utils/cpp_extension.py", line 2067, in _get_cuda_arch_flags
    supported_sm = [int(arch.split('_')[1])
                    ^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: '90a'
```

**Comment by Turakar** [comment]
Ah, it seems to be fixed in nightly, but not in 2.6.0.

**Pull request closed without merging**